### PR TITLE
fix: resolvers can be values rather than functions

### PR DESCRIPTION
**Pull request by rintaun**
Allows `Resolver` to be a value rather than just a function by adding `TResult` and `Promise<TResult>` as possible types.

Resolvers are generally functions, but they are also allowed to be values.

Related: #3112

### CODE REVIEW

**Comment by dotansimha** [comment]
@kamilkisiela can you please review?

File: packages/plugins/typescript/resolvers/src/index.ts [hunk 1]
```diff
@@ -105,15 +107,23 @@ export type StitchingResolver<TResult, TParent, TContext, TArgs> = {
   }
 
   if (noSchemaStitching) {
-    // Resolver = ResolverFn;
-    defsToInclude.push(`${resolverType} ${resolverFnUsage};`);
+    // Resolver =
+    // | TResult
+    // | Promise<TResult>
+    // | ResolverFn;
+    resolverDefParts.push(`  | ${resolverFnUsage};`);
```
Comment (dotansimha) [inline packages/plugins/typescript/resolvers/src/index.ts]: Why adding ` | ` to every string? You can add it without it, and then use `.join(' | ')`

Comment (rintaun) [reply packages/plugins/typescript/resolvers/src/index.ts]: That's a good point. Now that there are always multiple, it shouldn't be necessary anymore. I'll update the PR

Comment (dotansimha) [reply packages/plugins/typescript/resolvers/src/index.ts]: Thanks!
@kamilkisiela is the resolvers expert, he'll review it soon :)

**Comment by kamilkisiela** [comment]
I'm not quite sure GraphQL allows to use primitive values or Promises as resolvers. I agree they are correct when being returned via function and passed on to child types.

```graphql
type Story {
  id: ID!
  isManageable: Boolean!
}

type Query {
  stories: [Story!]
  story(id: ID!): Story
}
```

```typescript
interface Story {
  id: string;
  title: string;
  isManageable: (() => boolean);
}
const stories: Array<Story>;

const resolvers = {
  Query: {
    story(_, { id }) {
      return stories.find(u => u.id === id);
    },
    stories,
  }
} 
```


The `Query.story` resolver will work, `isManageable` will be resolved correctly but I don't thing `Query.stories` will return something non-null value. Correct me if I'm wrong.

The `Resolver` type produced by Codegen with your addition will allow to put strings, numbers, objects and arrays or even all of them wrapped with a Promise. 

What we expect is a resolver that returns an object with fields where each can be a primitive value, a function or a promise.

**Comment by rintaun** [comment]
I stand corrected. I don't know why I was under the impression that non-functional resolvers worked... Sorry for wasting your time 😬

**Pull request closed without merging**